### DISCUSSION TITLE
Add support for reading large files

### DISCUSSION THREAD

**Discussion by robin-nitrokey** [trussed-dev/trussed#48] [Ideas]
Currently, Trussed can only handle files up to `MAX_MESSAGE_LENGTH` = 1024 bytes long as the `write_file` and `read_file` methods of the [`FilesystemClient`](https://docs.rs/trussed/latest/trussed/client/trait.FilesystemClient.html) require and return the full content of the file.  This may not be enough in some use cases, e. g. some data objects in the OpenPGP card application.  Adding an API using offsets or a cursor could solve this problem and allow files with arbitrary length.

**Comment by arturkow2000** [comment]
@robin-nitrokey Passing an offset as argument could work when reading files. But for writing files Trussed should have an API that allows to open the file (returning handle/file descriptor), write data, flush/commit dat and close the handle.
Currently, a call to `write_file` causes Trussed to perform open/write/flush/close cycle. If `offset` were just added as argument each write would cause LittleFS to flush the changes to disk causing excessive I/O. Also, since LittleFS is copy-on-write, on each write would result in copying the data around.
